From b83203e47fca8b1675dda3d94c13dd5b6ded8a8d Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Tue, 22 Jan 2013 22:13:22 +0800
Subject: libstagefright: If nativewindow equals null, do not set scale mode

BZ: 81078

if nativewindow == NULL,native_window_set_scaling_mode will cause
TOMBSTONE error, add protection before native_window_set_scaling_mode

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: Icd94eaa129ff65f77a1bac9a17af18aa494b269e
Orig-Change-Id: I2b3af0a431d27baf22ce218fde5de9668a716182
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libstagefright/ACodec.cpp |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index e217168..b8ba036 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -3820,10 +3820,11 @@ bool ACodec::LoadedState::onConfigureComponent(
                 static_cast<NativeWindowWrapper *>(obj.get()));
         CHECK(nativeWindow != NULL);
         mCodec->mNativeWindow = nativeWindow->getNativeWindow();
-
-        native_window_set_scaling_mode(
-                mCodec->mNativeWindow.get(),
-                NATIVE_WINDOW_SCALING_MODE_SCALE_TO_WINDOW);
+        if (mCodec->mNativeWindow != NULL) {
+            native_window_set_scaling_mode(
+                    mCodec->mNativeWindow.get(),
+                    NATIVE_WINDOW_SCALING_MODE_SCALE_TO_WINDOW);
+        }
     }
     CHECK_EQ((status_t)OK, mCodec->initNativeWindow());
 
-- 
1.7.4.1

